a3p-integration/proposals/z:acceptance/psm.test.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * @file The goal of this file is to implement a set of tests to make sure PSM works properly
+ *
+ * Here are the steps we want to take;
+ * 1 - Change swap fees and mint limit according to "psmTestSpecs" below
+ * 2 - Create a new user using agd.keys
+ * 3 - Fund new user with a stable coin from the VALIDATOR
+ *     - Do not provision manually
+ * 4 - Make sure new user is able to mint IST from PSM (fees are applied)
+ * 5 - Make sure new user can pay their debt and get their anchor (fees are applied)
+ * 6 - Make sure mint limit is adhered
+ */
+
+import test from 'ava';
+import {
+  adjustBalancesIfNotProvisioned,
+  agopsPsm,
+  bankSend,
+  checkGovParams,
+  checkSwapExceedMintLimit,
+  checkSwapSucceeded,
+  checkUserInitializedSuccessfully,
+  getPsmMetrics,
+  implementPsmGovParamChange,
+  initializeNewUser,
+  maxMintBelowLimit,
+} from './test-lib/psm-lib.js';
+import {
+  getUser,
+  GOV1ADDR,
+  GOV2ADDR,
+  GOV3ADDR,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+import { getBalances } from './test-lib/utils.js';
+
+// Export these from synthetic-chain?
+export const USDC_DENOM = process.env.USDC_DENOM;
+export const PSM_PAIR = process.env.PSM_PAIR?.replace('.', '-');
+
+const psmTestSpecs = {
+  govParams: {
+    giveMintedFeeVal: 10n, // in %
+    wantMintedFeeVal: 10n, // in %
+    mintLimit: 500n * 1_000_000n, // in IST
+    deadline: 1, // in minutes
+  },
+  psmInstance: `psm-${PSM_PAIR}`,
+  // @ts-expect-error we assume PSM_PAIR is set because of synthetic-chain environment
+  anchor: PSM_PAIR.split('-')[1],
+  newUser: {
+    name: 'new-psm-trader',
+    fund: {
+      denom: USDC_DENOM ? USDC_DENOM : 'fake',
+      value: '300000000', // 300 USDC_grv
+    },
+  },
+  otherUser: {
+    name: 'gov1',
+    fund: {
+      denom: USDC_DENOM ? USDC_DENOM : 'fake',
+      value: '1000000000', // 1000 USDC_grv
+    },
+    toIst: {
+      value: 500, // in IST
+    },
+  },
+  toIst: {
+    value: 50, // in IST
+  },
+  fromIst: {
+    value: 50, // in USDC_axl
+  },
+};
+
+test.serial('change gov params', async t => {
+  await implementPsmGovParamChange(
+    {
+      address: GOV1ADDR,
+      instanceName: psmTestSpecs.psmInstance,
+      newParams: psmTestSpecs.govParams,
+      deadline: psmTestSpecs.govParams.deadline,
+    },
+    { committeeAddrs: [GOV1ADDR, GOV2ADDR, GOV3ADDR], position: 0 },
+  );
+
+  // Replace when https://github.com/Agoric/agoric-sdk/pull/10171 is in
```

Comment:
please land that first cc @Chris-Hibbert

Reply:
@Chris-Hibbert approved it yesterday => https://github.com/Agoric/agoric-sdk/pull/10171

So, as I understand, we should update this one with whatever is landed in #10171 then merge into master?

Reply:
Merge #10171 to master then rebase this onto master that has it.